service-registry/src/main/java/org/apache/servicecomb/serviceregistry/ServiceRegistry.java

```diff
@@ -59,4 +59,31 @@ MicroserviceInstances findServiceInstances(String appId, String microserviceName
   Microservice getRemoteMicroservice(String microserviceId);
 
   Features getFeatures();
+
+  /**
+   * <p>
+   * Register a third party service if not registered before, and add it's instances into
+   * {@linkplain org.apache.servicecomb.serviceregistry.consumer.StaticMicroserviceVersions StaticMicroserviceVersions}.
+   * </p>
+   * <p>
+   * The registered third party service has the same {@code appId} and {@code environment} as this microservice instance has,
+   * and there is only one schema represented by {@code schemaIntfCls}, whose name is the same as {@code microserviceName}.
+   * </p>
+   *
+   * @param microserviceName name of the 3rd party service, and this param also specifies the schemaId
+   * @param version version of this 3rd party service
+   * @param schemaIntfCls the producer interface of the service. This interface is used to generate swagger schema and
+   * can also be used for the proxy interface of RPC style invocation.
+   * @param instances the instances of this 3rd party service. Users only need to specify the endpoint information, other
+   * necessary information will be generate and set in the implementation of this method.
+   */
+  void registryMicroserviceMapping(String microserviceName, String version, Class<?> schemaIntfCls,
+      List<MicroserviceInstance> instances);
+
+  /**
+   * @see #registryMicroserviceMapping(String, String, Class, List)
+   * @param endpoints the endpoints of 3rd party service. Each of endpoints will be treated as a separated instance.
```

Comment:
add example for endpoint?

Reply:
Yes, the example is added in the comment of method `registerMicroserviceMappingByEndpoints`